src/Chainweb/Pact/PactService.hs

```diff
@@ -811,10 +813,11 @@ execHistoricalLookup bh (Domain' d) k = pactLabel "execHistoricalLookup" $ do
 
 execPreInsertCheckReq
     :: (CanReadablePayloadCas tbl, Logger logger)
-    => Vector ChainwebTransaction
+    => MemPoolAccess
+    -> Vector ChainwebTransaction
     -> PactServiceM logger tbl (Vector (Either Mempool.InsertError ChainwebTransaction))
-execPreInsertCheckReq txs = pactLabel "execPreInsertCheckReq" $ withDiscardedBatch $ do
-    let requestKeys = V.map P._cmdHash txs
+execPreInsertCheckReq mpAccess txs = pactLabel "execPreInsertCheckReq" $ withDiscardedBatch $ do
+    let requestKeys = V.map P.cmdToRequestKey txs
```

Comment:
Instead of calling into the mempool here, return `InsertError`s. You can add an extra case like `InsertErrorTimedOut`.